File: 08-2-1-SimpleFactory/Dice.cs
namespace SimpleFactory
{
    using System;

    public sealed class Dice
    {
        private readonly int _maxPips;
        private readonly Random _random;

        public Dice(int maxPips)
        {
            _maxPips = maxPips;
            _random = new Random();
        }

        public int Roll()
        {
            var roll = _random.Next(1, _maxPips + 1);

            return roll;
        }
    }

}

File: 08-DesignPatterns-Java/Starter/SimpleFactory/src/Dice.java
import java.util.Random;

public final class Dice
{
    private final int _maxPips;
    private final Random _random;

    public Dice(int maxPips)
    {
        _maxPips = maxPips;
        _random = new Random();
    }

    public int roll()
    {
        var roll = _random.nextInt(_maxPips + 1) + 1;

        return roll;
    }
}

